drivers/kscan/kscan_xpt2046.c

```diff
@@ -0,0 +1,291 @@
+	int32_t x = mx * meas.x + cx;
+
+	x = (x < 0 ? 0 : x) >> 16;
+
+	int32_t my = (config->screen_size_y << 16) / (config->max_y - config->min_y);
+	int32_t cy = (config->screen_size_y << 16) - my * config->max_y;
+	int32_t y = my * meas.y + cy;
+
+	y = (y < 0 ? 0 : y) >> 16;
+
+	bool pressed = meas.z > config->threshold;
+
+	/* Don't send any other than "pressed" events.
+	 * releasing seem to cause just random noise
+	 */
+	if (data->enabled && pressed) {
+		LOG_DBG("raw: x=%4u y=%4u ==> x=%4d y=%4d", meas.x, meas.y, x, y);
+		data->last_x = x;
+		data->last_y = y;
+		data->pressed = pressed;
+		data->callback(data->dev, (uint32_t)y, (uint32_t)x, pressed);
+		/* Ensure that we send released event */
+		k_work_reschedule(&data->dwork, K_MSEC(100));
+	}
+	gpio_add_callback(config->int_gpio.port, &data->int_gpio_cb);
+}
+
+static int xpt2046_configure(const struct device *dev, kscan_callback_t callback)
+{
+	struct xpt2046_data *data = dev->data;
+
+	if (!callback) {
+		LOG_ERR("Callback is null");
+		return -EINVAL;
+	}
+	LOG_DBG("%s: set callback", dev->name);
+
+	data->callback = callback;
+
+	return 0;
+}
+
+static int xpt2046_enable_callback(const struct device *dev)
+{
+	struct xpt2046_data *data = dev->data;
+	const struct xpt2046_config *config = dev->config;
+
+	LOG_DBG("%s: enable cb", dev->name);
+	data->enabled = true;
+	gpio_add_callback(config->int_gpio.port, &data->int_gpio_cb);
+	return 0;
+}
+
+static int xpt2046_disable_callback(const struct device *dev)
+{
+	struct xpt2046_data *data = dev->data;
+	const struct xpt2046_config *config = dev->config;
+
+	gpio_remove_callback(config->int_gpio.port, &data->int_gpio_cb);
+	data->enabled = false;
+
+	LOG_DBG("%s: disable cb", dev->name);
+	return 0;
+}
+
+static int xpt2046_init(const struct device *dev)
+{
+	int r;
+	const struct xpt2046_config *config = dev->config;
+	struct xpt2046_data *data = dev->data;
+
+	if (!spi_is_ready_dt(&config->bus)) {
+		LOG_ERR("SPI controller device not ready");
+		return -ENODEV;
+	}
+
+	data->dev = dev;
+	k_work_init(&data->work, xpt2046_work_handler);
+	k_work_init_delayable(&data->dwork, xpt2046_release_handler);
+
+	if (!gpio_is_ready_dt(&config->int_gpio)) {
+		LOG_ERR("Interrupt GPIO controller device not ready");
+		return -ENODEV;
+	}
+
+	r = gpio_pin_configure_dt(&config->int_gpio, GPIO_INPUT);
+	if (r < 0) {
+		LOG_ERR("Could not configure interrupt GPIO pin");
+		return r;
+	}
+
+	r = gpio_pin_interrupt_configure_dt(&config->int_gpio, GPIO_INT_EDGE_TO_ACTIVE);
+	if (r < 0) {
+		LOG_ERR("Could not configure interrupt GPIO interrupt.");
+		return r;
+	}
+
+	gpio_init_callback(&data->int_gpio_cb, xpt2046_isr_handler, BIT(config->int_gpio.pin));
+
+	LOG_INF("Init '%s' device", dev->name);
+
+	return 0;
+}
+
+static const struct kscan_driver_api xpt2046_driver_api = {
+	.config = xpt2046_configure,
+	.enable_callback = xpt2046_enable_callback,
+	.disable_callback = xpt2046_disable_callback,
+};
+
+#define DT_DRV_COMPAT xptek_xpt2046
```

Comment:
Please move it to the top of the file (as it's usually done in the rest of drivers).

Reply:
Fixed.